### DISCUSSION TITLE
在linux上make安装，采用protobuf-c1.2.1和protobuf-3.5.1总是会报c++错误

### DISCUSSION THREAD

**Discussion by jeasnd** [protobuf-c/protobuf-c#712] [Q&A]
root@ba56f6a3b040:~/workspace/test/protobuf-c-1.2.1# make
  CXX      protoc-c/protoc_c_protoc_c-c_bytes_field.o
In file included from protoc-c/c_bytes_field.cc:63:0:
./protoc-c/c_bytes_field.h:89:3: error: ‘map’ does not name a type
   map<string, string> variables_;
   ^
protoc-c/c_bytes_field.cc:77:25: error: ‘map’ has not been declared
                         map<string, string>* variables) {
                         ^
protoc-c/c_bytes_field.cc:77:28: error: expected ‘,’ or ‘...’ before ‘<’ token
                         map<string, string>* variables) {
                            ^
protoc-c/c_bytes_field.cc: In function ‘void google::protobuf::compiler::c::SetBytesVariables(const google::protobuf::FieldDescriptor*, int)’:
protoc-c/c_bytes_field.cc:78:5: error: ‘variables’ was not declared in this scope
   (*variables)["name"] = FieldName(descriptor);
     ^
protoc-c/c_bytes_field.cc: In constructor ‘google::protobuf::compiler::c::BytesFieldGenerator::BytesFieldGenerator(const google::protobuf::FieldDescriptor*)’:
protoc-c/c_bytes_field.cc:89:34: error: ‘variables_’ was not declared in this scope
   SetBytesVariables(descriptor, &variables_);
                                  ^
protoc-c/c_bytes_field.cc: In member function ‘virtual void google::protobuf::compiler::c::BytesFieldGenerator::GenerateStructMembers(google::protobuf::io::Printer*) const’:
protoc-c/c_bytes_field.cc:101:22: error: ‘variables_’ was not declared in this scope
       printer->Print(variables_, "ProtobufCBinaryData $name$$deprecated$;\n");
                      ^
protoc-c/c_bytes_field.cc: In member function ‘virtual void google::protobuf::compiler::c::BytesFieldGenerator::GenerateStaticInit(google::protobuf::io::Printer*) const’:
protoc-c/c_bytes_field.cc:142:22: error: ‘variables_’ was not declared in this scope
       printer->Print(variables_, "$default_value$");
                      ^
Makefile:1172: recipe for target 'protoc-c/protoc_c_protoc_c-c_bytes_field.o' failed
make: *** [protoc-c/protoc_c_protoc_c-c_bytes_field.o] Error 1

**Comment by Hming1** [comment]
Did you solve this problem, I was in the same situation, can you share that? Thank you!
